### DISCUSSION TITLE
Canvas Embed

### DISCUSSION THREAD

**Discussion by uroybd** [oleeskild/obsidian-digital-garden#225] [Ideas]
I would like to implement the Canvas embed in a chunk of free time I might get in April. Here is the outline of what I'm thinking about it.

**The feature will work for canvases embedded in a note.**

I have two strategies in mind and am kind of confused about the best route to go.

## Strategy 1: Thin Embed.

1. Plugin will do the minimal transformation to the canvas JSON. (e.g. changing Paths to garden links, upload embedded images) and then output it like this:
```
```canvasjson
{
...
}
```

2. A lightweight js library will generate the canvas using that data. It will use iframes for loading notes on demand to load the notes.

## Strategy 2: Complete Embed
It will transclude everything in the note. Problem is, it will make the pages heavier.

**Comment by oleeskild** [comment]
Supporting canvas would be really cool! I think strategy 1 would be the best alternative of the two. It would behave similar to what you would expect. If any of the linked notes changes, the canvas will change to, without having to republish it. And if the note in the canvas isn't published, it shouldn't be viewable in the canvas. We get a lot of logic "for free". 

It would also be a lighter calculation to figure out whether it has changed or not in the publication center.
